packages/core/src/api/channel.ts

```diff
@@ -0,0 +1,161 @@
+import { makeURLSearchParams, type REST } from '@discordjs/rest';
+import {
+	Routes,
+	type APIChannel,
+	type APIFollowedChannel,
+	type APIInvite,
+	type APIMessage,
+	type RESTGetAPIChannelUsersThreadsArchivedResult,
+	type RESTPatchAPIChannelJSONBody,
+	type RESTPostAPIChannelInviteJSONBody,
+	type RESTGetAPIChannelMessagesQuery,
+	type RESTGetAPIChannelThreadsArchivedQuery,
+} from 'discord-api-types/v10';
+
+export class ChannelsAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Fetches a channel
+	 *
+	 * @param channelId - The id of the channel
+	 */
+	public async get(channelId: string) {
+		return (await this.rest.get(Routes.channel(channelId))) as APIChannel;
+	}
+
+	/**
+	 * Edits a channel
+	 *
+	 * @param channelId - The id of the channel to edit
+	 * @param channel - The new channel data
+	 */
+	public async edit(channelId: string, channel: RESTPatchAPIChannelJSONBody) {
+		return (await this.rest.patch(Routes.channel(channelId), {
+			body: channel,
+		})) as APIChannel;
+	}
+
+	/**
+	 * Deletes a channel
+	 *
+	 * @param channelId - The id of the channel to delete
+	 */
+	public async delete(channelId: string) {
+		return (await this.rest.delete(Routes.channel(channelId))) as APIChannel;
+	}
+
+	/**
+	 * Fetches the messages of a channel
+	 *
+	 * @param channelId - The id of the channel to fetch messages from
+	 * @param options - The options to use when fetching messages
+	 */
+	public async getMessages(channelId: string, options: RESTGetAPIChannelMessagesQuery = {}) {
+		return (await this.rest.get(Routes.channelMessages(channelId), {
+			query: makeURLSearchParams(options as Record<string, unknown>),
+		})) as APIMessage[];
+	}
+
+	/**
+	 * Shows a typing indicator in a channel
+	 *
+	 * @param channelId - The id of the channel to show the typing indicator in
+	 */
+	public async showTyping(channelId: string) {
+		await this.rest.post(Routes.channelTyping(channelId));
+	}
+
+	/**
+	 * Fetches the pinned messages of a channel
+	 *
+	 * @param channelId - The id of the channel to fetch pinned messages from
+	 */
+	public async getPins(channelId: string) {
+		return (await this.rest.get(Routes.channelPins(channelId))) as APIMessage[];
+	}
+
+	/**
+	 * Pins a message in a channel
+	 *
+	 * @param channelId - The id of the channel to pin the message in
+	 * @param messageId - The id of the message to pin
+	 */
+	public async pin(channelId: string, messageId: string) {
```

Comment:
```suggestion
	public async pinMessage(channelId: string, messageId: string) {
```
We don't pin a channel